crewai_tools/tools/apify_tool/apify_tool.py

```diff
@@ -0,0 +1,71 @@
+import typing as t
+from crewai_tools.tools.base_tool import BaseTool
+from apify_client import ApifyClient, ActorRun, Dataset
+
+class ApifyTool(BaseTool):
+    def __init__(self, apify_token: str):
+        self.client = ApifyClient(apify_token)
+
+    def _run(self, query: str) -> str:
+        """
+        Run the Apify tool with the given query.
+        """
+        # Parse the query and determine the appropriate Apify action
+        # For example, you could use natural language processing to extract the intent
+        # and map it to an Apify action
+
+        # Execute the Apify action using the client
+        # For example, to run an actor:
+        actor_run = self.run_actor(actor_id="your_actor_id", run_input={"query": query})
```

Comment:
I think we want the agent to pass the actor id in this case together with the query?
Also might be worth having another tool to lsit the actor ids so it can decide / learn what to call?